client/me/two-step/index.jsx

```diff
@@ -39,6 +40,24 @@ class TwoStep extends Component {
 		this.props.fetchUserSettings();
 	};
 
+	componentDidMount() {
+		if ( ! this.props.isFetchingUserSettings && ! this.props.isEmailVerified ) {
+			window.location.href = '/me/security';
+		}
+	}
+
+	componentDidUpdate( prevProps ) {
+		// Check if fetching just completed
+		if (
+			prevProps.isFetchingUserSettings &&
+			! this.props.isFetchingUserSettings &&
+			prevProps.isEmailVerified &&
+			! this.props.isEmailVerified
```

Comment:
If the fetching is done, I think we should just check the final `isEmailVerified` value, right? Or does it start as `true`?

Reply:
I would do `this.props.isEmailVerified === false` and get rid of all the other checks.

Reply:
Changed